### DISCUSSION TITLE
TextMeshPro support in AnKuchen Template

### DISCUSSION THREAD

**Discussion by slimshader** [kyubuns/AnKuchen#13] [Q&A] [answered]
Hi,

When generating AnKuchen template TMP_Text and TMP_InputField components are missing in the generated code. I discovered that I can create AnKuchenCopyTemplateSettings so I added "TMP_Text" and "TMP_InputField" to the PickupComponentNames section but it didn't help. Is it possible to get TMP components generated in the Template?

**Comment by kyubuns** [comment, marked as answer]
Hi!

PickupComponentNames must match the class name exactly.
You should specify `TextMeshProUGUI` and `TMP_InputField`.

Currently, this does not even check the name of the class or interface from which it is inherited.
After listening to you, I felt it would be useful to include this support, so I will do so in a near future update.

**Reply by kyubuns** [reply]
In v1.8.2, which was released today, the names of interface and base class are also checked during Copy Template.
This should allow you to get the elements as expected by simply specifying "TMP_Text" and "TMP_InputField".
